### ISSUE TITLE
[Question] How are the rules numbered?

### ISSUE THREAD

**Issue by zabio3** [hadolint/hadolint#305]
- [x] I searched existing issues before opening this one

I'm a bit confused about the DL rule numbering.

I want to know how to assign the rule number.

 - The next rule of DL3026 is DL4000, but what is the reason for moving from 3000s to 4000s?

 - Is there a reason why DL4002 is missing?

**Comment by lorenzo** [comment]
I'm not sure about the historic reasons for this, I took ownership of the project after those decision were made. Here's what I can infer:

- `DL3000-DL3999`: Used for rules that are less opinionated, are present in the docker styleguido or are actual errors
- `DL4000-DL4999`: It seems like this was used for minor warnings, and things related to style or deprecated things. Unfortunately I added important rules to this range without noticing there was a pattern before. All the rule I added to this rage where related to `SHELL` related things, for example rules that could actually be par of SchellCheck

`DL4002` seems to have been skipped by mistake. This is the commit where it happened https://github.com/hadolint/hadolint/commit/0e9e0a75ee0fc243601ad9d6f74330048c064126#diff-04c6e90faac2675aa89e2176d2eec7d8
